plugins/java-decompiler/engine/src/org/jetbrains/java/decompiler/main/InitializerProcessor.java

```diff
@@ -139,7 +139,8 @@ private static void extractStaticInitializers(ClassWrapper wrapper, MethodWrappe
             if (fexpr.isStatic() && fexpr.getClassname().equals(cl.qualifiedName) &&
                 cl.hasField(fexpr.getName(), fexpr.getDescriptor().descriptorString)) {
 
-              if (isExprentIndependent(asexpr.getRight(), meth)) {
+              // interfaces fields should always be initialized inline
+              if (isInterface || isExprentIndependent(asexpr.getRight(), meth)) {
```

Comment:
On the one hand, this is correct. On the other hand, decompiler often deals with a weird bytecode (obfuscated, instrumented, non-Java etc). In your example, `isExprentIndependent()` should have decided that the initializer is independent and could be inlined - so that a "normal" code would decompile into a correct source, and a "weird" code would produce something incorrect but at least understandable.